frontend/sozisel-app/src/components/TemplatesList/TemplateList.tsx

```diff
@@ -0,0 +1,146 @@
+import "./TemplateList.scss";
+
+import MuiAlert, { AlertProps } from "@material-ui/lab/Alert";
+import {
+  SessionTemplate,
+  useCloneSessionTemplateMutation,
+  useDeleteSessionTemplateMutation,
+  useSearchSessionTemplatesQuery,
+} from "../../graphql";
+
+import CircularProgress from "@material-ui/core/CircularProgress";
+import List from "@material-ui/core/List";
+import MainNavbar from "../MainNavbar/MainNavbar";
+import { ReactElement } from "react";
+import SearchBar from "./SearchBar/SearchBar";
+import Snackbar from "@material-ui/core/Snackbar";
+import TemplateCard from "./TemplateCard/TemplateCard";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+function Alert(props: AlertProps) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
+export default function TemplateList(): ReactElement {
+  const { t } = useTranslation("common");
+  const { data, loading, error, refetch } = useSearchSessionTemplatesQuery({
+    fetchPolicy: "network-only",
+  });
+  const [name, setName] = useState<string>("");
+  const [includePublic, setIncludePublic] = useState<boolean>(false);
+  const [
+    cloneMutation,
+    { error: cloneError, loading: cloneLoading },
+  ] = useCloneSessionTemplateMutation();
+  const [
+    deleteMutation,
+    { error: deleteError, loading: deleteLoading },
+  ] = useDeleteSessionTemplateMutation();
+  const [successMsg, setSuccessMsg] = useState<string>("");
+
+  const onSearch = ({
+    nameSearch = name,
+    includePublicSearch = includePublic,
+  }: {
+    nameSearch?: string;
+    includePublicSearch?: boolean;
+  }) => {
+    refetch({ name: nameSearch, includePublic: includePublicSearch });
+    setName(nameSearch);
+    setIncludePublic(includePublicSearch);
+  };
+
+  const onCopy = async (template: SessionTemplate) => {
```

Comment:
As those functions are passed down the component (used in nested component) put those functions inside `useCallback` to avoid unnecessary rerenders.